packet/bgp/bgp.go

```diff
@@ -7982,6 +7982,33 @@ func (msg *BGPUpdate) IsEndOfRib() (bool, RouteFamily) {
 	return false, RouteFamily(0)
 }
 
+func TreatAsWithdraw(msg *BGPUpdate) *BGPUpdate {
+	withdraw := &BGPUpdate{
+		WithdrawnRoutesLen:    0,
+		WithdrawnRoutes:       []*IPAddrPrefix{},
+		TotalPathAttributeLen: 0,
+		PathAttributes:        make([]PathAttributeInterface, 0, len(msg.PathAttributes)),
+		NLRI:                  []*IPAddrPrefix{},
+	}
+	withdraw.WithdrawnRoutes = append(msg.WithdrawnRoutes, msg.NLRI...)
+	var unreach []AddrPrefixInterface
+
+	for _, p := range msg.PathAttributes {
+		switch nlri := p.(type) {
+		case *PathAttributeMpReachNLRI:
+			unreach = append(unreach, nlri.Value...)
+		case *PathAttributeMpUnreachNLRI:
+			unreach = append(unreach, nlri.Value...)
+		default:
+			withdraw.PathAttributes = append(withdraw.PathAttributes, p)
```

Comment:
Why necessary? Withdraw message doesn't need any attributes.

Reply:
It is not necessary, sorry.
I will remove it.